fsm/fsm_group.go

```diff
@@ -43,6 +44,16 @@ func (s *stateGroup) IsTerminated(out StateType) bool {
 	return s.d.reachedFinalityState(out)
 }
 
+// GetSync will make sure all events present at the time of the call are processed before
+// returning a value, which is read into out
+func (s *stateGroup) GetSync(ctx context.Context, id interface{}, out cbg.CBORUnmarshaler) error {
+	err := s.SendSync(ctx, id, nilEvent{})
+	if err != nil {
+		return err
+	}
+	return s.Get(id).Get(out)
```

Comment:
Was wondering if there was a chance for a race here?  I'm not sure how `GetSync` gets called in the loop; is it possible that more events have come in after the `nilEvent`, between the `SendSync` and `Get` calls?  Not sure that it matters for our use case.

Reply:
yes, it is technically possible. I'm not concerned. The main concern is anything that happened before GetSync was called is processed. That's the guarantee GetSync offers.